backend/sirius-components-diagrams-layout/src/test/java/org/eclipse/sirius/components/diagrams/layout/incremental/BorderNodePositionTests.java

```diff
@@ -0,0 +1,247 @@
+/*******************************************************************************
+ * Copyright (c) 2022 THALES GLOBAL SERVICES.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.diagrams.layout.incremental;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.eclipse.elk.alg.layered.options.LayeredOptions;
+import org.eclipse.elk.core.data.LayoutMetaDataService;
+import org.eclipse.sirius.components.diagrams.NodeType;
+import org.eclipse.sirius.components.diagrams.Position;
+import org.eclipse.sirius.components.diagrams.Size;
+import org.eclipse.sirius.components.diagrams.TextBounds;
+import org.eclipse.sirius.components.diagrams.events.IDiagramEvent;
+import org.eclipse.sirius.components.diagrams.events.MoveEvent;
+import org.eclipse.sirius.components.diagrams.events.ResizeEvent;
+import org.eclipse.sirius.components.diagrams.layout.LayoutConfiguratorRegistry;
+import org.eclipse.sirius.components.diagrams.layout.incremental.data.DiagramLayoutData;
+import org.eclipse.sirius.components.diagrams.layout.incremental.data.IContainerLayoutData;
+import org.eclipse.sirius.components.diagrams.layout.incremental.data.LabelLayoutData;
+import org.eclipse.sirius.components.diagrams.layout.incremental.data.NodeLayoutData;
+import org.eclipse.sirius.components.diagrams.layout.incremental.provider.ImageSizeProvider;
+import org.eclipse.sirius.components.diagrams.layout.incremental.provider.NodeSizeProvider;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test cases for border nodes.
+ *
+ * @author lfasani
+ */
+class BorderNodePositionTests {
+
+    private static final Size DEFAULT_NODE_SIZE = Size.of(200, 100);
+
+    private static final Size DEFAULT_BORDER_NODE_SIZE = Size.of(24, 20);
+
+    private static final Position ZERO_POSITION = Position.at(0, 0);
+
+    @BeforeAll
+    private static void initTest() {
+        LayoutMetaDataService.getInstance().registerLayoutMetaDataProviders(new LayeredOptions());
+    }
+
+    /**
+     * Test the snap of the border nodes on their parent container.<br/>
+     * The border nodes are voluntary at an invalid position to make sure they properly snap done at the first layout
+     */
+    @Test
+    void testSnap() {
```

Comment:
why the lack of visibility on this method and the other methods?

Reply:
because it was adviced by checkstyle

Reply:
I put the public visibility back